app/src/pages/p/[projectSlug]/usage/index.tsx

```diff
@@ -16,72 +16,240 @@ import {
   Tr,
   VStack,
   Box,
+  Spacer,
+  Tabs,
+  TabList,
+  Tab,
+  TabPanels,
+  TabPanel,
 } from "@chakra-ui/react";
-import { DollarSign, Hash } from "lucide-react";
+import { ChevronRightIcon, ChevronLeftIcon, DollarSign, Hash } from "lucide-react";
+import { useRouter } from "next/router";
+import { useEffect, useMemo } from "react";
+import { useQueryParam, StringParam } from "use-query-params";
 import { ProjectLink } from "~/components/ProjectLink";
+import CostGraph from "~/components/dashboard/CostGraph";
 
 import UsageGraph from "~/components/dashboard/UsageGraph";
 import AppShell from "~/components/nav/AppShell";
 import { modelInfo } from "~/server/fineTuningProviders/supportedModels";
-import { api } from "~/utils/api";
-import dayjs from "~/utils/dayjs";
-import { useSelectedProject } from "~/utils/hooks";
+import dayjs, { formatToUTCDayMonth } from "~/utils/dayjs";
+import { useSelectedProject, useStats } from "~/utils/hooks";
 
 const numberWithDefault = (num: number | string | bigint | null, defaultValue = 0) =>
   Number(num ?? defaultValue);
 
 export default function Usage() {
   const { data: selectedProject } = useSelectedProject();
 
-  const stats = api.usage.stats.useQuery(
-    { projectId: selectedProject?.id ?? "" },
-    { enabled: !!selectedProject },
-  );
+  const router = useRouter();
+
+  const [startDate, setStartDate] = useQueryParam("start", StringParam);
+  const [endDate, setEndDate] = useQueryParam("end", StringParam);
+
+  const startOfThisMonth = dayjs().startOf("month").format("YYYY-MM-DD");
+  const endOfThisMonth = dayjs().endOf("month").format("YYYY-MM-DD");
+
+  const { projectSlug } = router.query;
+
+  useEffect(() => {
+    // Prevents `href` Interpolation error in nextjs
+    if (projectSlug) {
```

Comment:
What does this check do? Or alternatively, under what circumstances would `projectSlug` be false? Can we accomplish the same thing with `selectedProject.id`?

Reply:
Additionally, by using the `withDefault` helper from `useQueryParams` you can probably get rid of this `useEffect` entirely by just setting the default values of the `startDate`/`endDate`? I think it's actually better to show the current month by default when you load the page without that being explicit in the URL, and only update the URL query string if you navigate to non-default dates.

Reply:
Managed to remove all of that by using useQueryParams.